report_barcode_elaphe/controllers/main.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+# Odoo, an open source suite of business apps
+# This module copyright (C) 2015 bloopark systems (<http://bloopark.de>).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp.addons.report.controllers.main import ReportController
+from openerp.addons.web.http import route, request
+
+import cStringIO
+import re
+
+from werkzeug import exceptions
+
+try:
+    from elaphe import barcode
+except Exception:
+    pass
+
+
+class ReportBarcodeController(ReportController):
+
+    """The class of Report controller"""
+
+    @route()
+    def report_barcode(self, type, value, width=0, height=0,
+                       humanreadable=0, **kw):
```

Comment:
Completely re-implementing a method is not desirable in a lot of instances, particularly when unknown modules could depend on, or add to via inheritance, implementation on the parent. You can't guarantee module load order, so inconsistent results are common if not calling the superclass.

I would prefer if there were an argument to explicitly use this implementation & send to super if not defined. This will guard against unpredictable behavior while still allowing for the fanciness that this module provides 😄